internal/discovery/seed.go

```diff
@@ -36,6 +36,7 @@ func NewGrpcSeed(maddr ma.Multiaddr, node *locators.Node,
 }
 
 type hostService struct {
+	pb.UnimplementedDiscoveryServiceServer
```

Comment:
Why it's not implemented if `discovery.hostService` implements `DiscoveryServiceServer`?

Reply:
It's required for server to embed `Unimplemented***ServiceServer` struct.
This requirement guarantees that backward-compatibility is maintained by the generated code when new methods are added to a service, which is, itself, a backward-compatible operation.
It was [implemented](https://github.com/grpc/grpc-go/releases/tag/cmd%2Fprotoc-gen-go-grpc%2Fv1.0.0) in v1.0.0 of `protoc-gen-go-grpc'.
We can avoid it by setting the corresponding flag at the generation, but it's not recommended.

Reply:
OK, I see, then it's fine.